lib/c-api/tests/deprecated/test-module-serialize.c

```diff
@@ -28,18 +28,6 @@ int main()
     printf("Serialized module pointer: %p\n", serialized_module_bytes.bytes);
     printf("Serialized module length: %d\n", serialized_module_bytes.bytes_len);
     assert(serialized_module_bytes.bytes != NULL);
-    assert(serialized_module_bytes.bytes_len > 11);
-    assert(serialized_module_bytes.bytes[0] == '\0');
-    assert(serialized_module_bytes.bytes[1] == 'w');
-    assert(serialized_module_bytes.bytes[2] == 'a');
-    assert(serialized_module_bytes.bytes[3] == 's');
-    assert(serialized_module_bytes.bytes[4] == 'm');
-    assert(serialized_module_bytes.bytes[5] == 'e');
-    assert(serialized_module_bytes.bytes[6] == 'r');
-    assert(serialized_module_bytes.bytes[7] == '-');
-    assert(serialized_module_bytes.bytes[8] == 'j');
-    assert(serialized_module_bytes.bytes[9] == 'i');
-    assert(serialized_module_bytes.bytes[10] == 't');
```

Comment:
Since this is the deprecated API it may not be worth testing, but it'd be good to have some logic here for verifying the binary blob we created isn't garbage.

I don't know if there's enough commonality to make that work... It might also make sense to have `CMake` able to pass defines to the C source so that the C code being compiled can be aware of this kind of thing and make adjustments as necessary.

Actually if we're testing that it deserializes, then it's probably sufficiently well tested.

Reply:
> Actually if we're testing that it deserializes, then it's probably sufficiently well tested.

Yep, that's exactly the thought process I went through.